dataflow/src/node/process.rs

```diff
@@ -287,6 +259,34 @@ impl Node {
     }
 }
 
+// When we miss at a can_query_through node we have to replay from
+// its parent instead of that node, so we'll update the misses:
+fn reroute_miss(nodes: &DomainNodes, miss: &mut Miss) {
+    let node = nodes[&miss.on].borrow();
+    if node.is_internal() && node.can_query_through() {
+        for col in miss.lookup_idx.iter_mut() {
+            let parents = node.resolve(*col).unwrap();
+            assert_eq!(
+                parents.len(),
+                1,
+                "can't query_through with more than one parent"
+            );
+
+            let (parent, parent_col) = parents[0];
+            let parent_node = nodes
+                .values()
+                .find(|n| n.borrow().global_addr() == parent)
+                .unwrap();
+
+            miss.on = *parent_node.borrow().local_addr();
```

Comment:
Why is this *inside* the `for` loop? I'd put it outside and assert that all the columns you resolve resolve to the same parent node.